classes/Tools.php

```diff
@@ -3158,11 +3158,17 @@ public static function clearSf2Cache($env = null)
             $env = _PS_MODE_DEV_ ? 'dev' : 'prod';
         }
 
-        $dir = _PS_ROOT_DIR_ . '/var/cache/' . $env . '/';
-
-        register_shutdown_function(function() use ($dir) {
-            $fs = new Filesystem();
-            $fs->remove($dir);
+        register_shutdown_function(function() use ($env) {
+            $directory = _PS_ROOT_DIR_ . '/var/cache/' . $env . '/';
+                if (is_dir($directory)) {
+                $directories = new DirectoryIterator($directory);
+                $fs = new Filesystem();
+                foreach ($directories as $fileinfo) {
+                    if (!$fileinfo->isDot() && $fileinfo->getFilename() != 'smarty') {
+                        $fs->remove($fileinfo->getPathName());
```

Comment:
Have you considered the use of `Finder` (https://symfony.com/doc/current/components/finder.html) with the method `notName()`?